governance/handler.go

```diff
@@ -44,67 +44,58 @@ var (
 )
 
 var GovernanceItems = map[int]check{
-	params.GovernanceMode:          {stringT, checkGovernanceMode, updateGovernanceConfig},
-	params.GoverningNode:           {addressT, checkAddress, updateGovernanceConfig},
-	params.UnitPrice:               {uint64T, checkUint64andBool, updateGovernanceConfig},
-	params.AddValidator:            {addressT, checkAddress, updateGovernanceConfig},
-	params.RemoveValidator:         {addressT, checkAddress, updateGovernanceConfig},
-	params.MintingAmount:           {stringT, checkBigInt, updateGovernanceConfig},
-	params.Ratio:                   {stringT, checkRatio, updateGovernanceConfig},
-	params.UseGiniCoeff:            {boolT, checkUint64andBool, updateGovernanceConfig},
-	params.DeferredTxFee:           {boolT, checkUint64andBool, updateGovernanceConfig},
-	params.MinimumStake:            {stringT, checkBigInt, updateGovernanceConfig},
-	params.StakeUpdateInterval:     {uint64T, checkUint64andBool, updateGovernanceConfig},
-	params.ProposerRefreshInterval: {uint64T, checkUint64andBool, updateGovernanceConfig},
-	params.Epoch:                   {uint64T, checkUint64andBool, updateGovernanceConfig},
-	params.Policy:                  {uint64T, checkUint64andBool, updateGovernanceConfig},
-	params.CommitteeSize:           {uint64T, checkUint64andBool, updateGovernanceConfig},
-	params.ConstTxGasHumanReadable: {uint64T, checkUint64andBool, updateParams},
+	params.GovernanceMode:          {stringT, checkGovernanceMode, nil},
```

Comment:
Is there a criterion for when I need to handle a trigger?

Reply:
@jimni1222 This map defines 
1. Governance Item (as a key of the map)
2. the key's data type
3. the validator function for the type
4. trigger function which to be called when the value of the key changed if it requires more action other than storing it in `governance.currentSet`

So it is triggered when the change have to affect an object other than `governance`

Reply:
Ah, ! that means, if update need to trigger something more than action in governance.currentSet, then trigger should be defined ! Thank you for explain :)